### PR TITLE
FOCUS #231: ListUnitPrice metric review

### PR DESCRIPTION
**Pull request by ijurica**
- Normative paragraph adjustment
- Supporting content - Providers mappings adjustments

### CODE REVIEW

File: supporting_content/metrics/listunitprice.md [hunk 1]
```diff
@@ -6,72 +6,71 @@ Current column mappings found in available data sets:
...
 
 ## Example usage scenarios
 
-- See [Appendix: Quantity-Driven Pricing strategies](../apendix/quantity_driven_pricing_strategies.md) section for more information on
-  - Quantity-Driven Pricing strategies (such as flat-rate, volume-based rates and tier-based rating)
-  - related UC scenarios
-  - Current values observed in billing data for various scenarios
-  - Alternative data sources for various scenarios
-- See [Pricing Support – UCs and Data samples Spreadsheet](https://docs.google.com/spreadsheets/d/1AZ-vtkKeKwYc8rqhxP1zMTnAVAS-svmWQQmr8cpv-IM/edit#gid=117987709) for additional UC scenarios and sample data.
+* See [Appendix: Quantity-Driven Pricing strategies](../apendix/quantity_driven_pricing_strategies.md) section for more information on
```
Comment (mattray) [inline supporting_content/metrics/listunitprice.md:39]: that should be "./appendix/quantity_driven_pricing_strategies.md" (note "appendix")

File: specification/metrics/listunitprice.md [hunk 2]
```diff
@@ -1,8 +1,8 @@
 # List Unit Price
```
Comment (yprikhodko) [inline specification/metrics/listunitprice.md:1]: Practitioners can calculate this by dividing List Cost to usage amount. Unless I'm missing something this column might be considered redundant. Happy to discuss further this topic with the group.

**Comment by nfonrose** [comment]
I would update the first sentence to :

`The List Unit Price represents a suggested provider published unit price for a single pricing measurement unit (identified by a [SkuId](#skuid)), which incorporates volume/tier-based rates while excluding any negotiated or commitment-based discounts.`

I've added the "(identified by a [SkuId](#skuid))" part

**Review by mattray — APPROVED** [review]

File: specification/metrics/listunitprice.md [hunk 3]
```diff
@@ -1,8 +1,8 @@
...
 
-The List Unit Price represents the suggested provider published unit price for a single pricing measurement unit, which incorporates volume/tier-based rates while excluding any negotiated or commitment-based discounts. This price is denominated in the [Billing Currency](#billingcurrency). It is used to calculate List Cost and savings based on rate optimization activities.
+The List Unit Price represents a suggested provider published unit price for a single pricing measurement unit, which incorporates volume/tier-based rates while excluding any negotiated or commitment-based discounts. This price is denominated in the [Billing Currency](#billingcurrency). It is used to calculate List Cost and savings based on rate optimization activities.
```
Comment (yprikhodko) [inline specification/metrics/listunitprice.md:3]: Should be in sync with the [suggestion](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/235/files#r1370605836) :
...while excluding any negotiated, commitment-based discounts or spare capacity (spot) discounts...

File: specification/metrics/listunitprice.md [hunk 4]
```diff
@@ -1,8 +1,8 @@
 # List Unit Price
 
-The List Unit Price represents the suggested provider published unit price for a single pricing measurement unit, which incorporates volume/tier-based rates while excluding any negotiated or commitment-based discounts. This price is denominated in the [Billing Currency](#billingcurrency). It is used to calculate List Cost and savings based on rate optimization activities.
+The List Unit Price represents a suggested provider-published unit price for a single pricing measurement unit (identified by a [Sku Price Id](#skupriceid)), which incorporates volume/tier-based rates while excluding any negotiated or commitment-based discounts. This price is denominated in the [Billing Currency](#billingcurrency). It is used to calculate List Cost and savings based on rate optimization activities.
```
Comment (ijurica) [inline specification/metrics/listunitprice.md:3]: in response to the [suggestion to add _...(identified by SkuId)..._](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/234#issuecomment-1777791478) - discussed and agreed to add _...(identified by Sku Price Id)..._

File: specification/metrics/listunitprice.md (hunk 3 above)
Comment (ijurica) [reply specification/metrics/listunitprice.md]: To include the capability to calculate savings from utilizing spare capacity (spot) discounts, we are still exploring a couple of alternative List Unit Price definitions.

- OPTION A (current version): _The List Unit Price represents a suggested provider-published unit price for a single pricing measurement unit (identified by a [Sku Price Id](#skupriceid)), which incorporates volume/tier-based rates while excluding any negotiated discounts, commitment-based discounts or the amortization of upfront charges (one-time or recurring)._

- OPTION B : _The List Unit Price represents a suggested provider-published unit price for a single pricing measurement unit (identified by a [Sku Price Id](#skupriceid)), which incorporates volume/tier-based rates while excluding any negotiated discounts, commitment-based discounts, **spare capacity (spot) discounts** or the amortization of upfront charges (one-time or recurring)._

- OPTION C (aligned with pricing model column terminology): _The List Unit Price represents a suggested provider-published unit price for a single pricing measurement unit (identified by a [Sku Price Id](#skupriceid)), which incorporates volume/tier-based rates while excluding any negotiated discounts, commitment-based discounts, **the impacts of dynamic pricing model** or the amortization of upfront charges (one-time or recurring)._

Note: **Alignment of all unit price and cost metric definitions is implied.**

Comment (ljadvey) [reply specification/metrics/listunitprice.md]: B is most clear but lacks flexibility of C.  Instead of "the impacts of dynamic pricing model" how about "dynamic pricing models (e.g. spare capacity (spot) discounts)"?

Comment (udam-f2) [reply specification/metrics/listunitprice.md]: I think we first need to determine if ListCost can/should behave in this way (excluding dynamic prices and replacing with 'fixed' prices of a similar (not equivalent) SKU. Once we have that decision (poll wording is in review), we should decide how to write this?

File: specification/metrics/listunitprice.md [hunk 5]
```diff
@@ -1,8 +1,8 @@
...
+The List Unit Price represents a suggested provider-published unit price for a single pricing measurement unit (identified by a [Sku Price Id](#skupriceid)), which incorporates free-tier, volume/tier-based, BYOL-based and dynamically priced SKU rates, while excluding any other discounts or the amortization of upfront charges (one-time or recurring). This price is denominated in the [Billing Currency](#billingcurrency). The List Unit Price can be used for calculating savings based on rate optimization activities. In this context, activities such as increased consumption leading to lower volume/tier-based rates, the use of pre-owned software licenses (BYOL - Bring Your Own License), leveraging interruptible resources and/or services, or optimizing usage to take advantage of dynamic pricing models are not considered rate optimization activities.
```
Comment (ijurica) [inline specification/metrics/listunitprice.md:3]: The new version of the definition requires review.

File: specification/metrics/listunitprice.md (hunk 3 above)
Comment (ijurica) [reply specification/metrics/listunitprice.md]: Based on the further in-meeting discussion, a new version of the definition has been prepared. 
Closing this conversation and opening a related conversation available [here](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/234#discussion_r1382558297).

File: specification/metrics/listunitprice.md (hunk 5 above)
Comment (udam-f2) [inline specification/metrics/listunitprice.md:3]:
```suggestion
The List Unit Price represents a suggested provider-published unit price, identified by the [SKU Price ID](#skupriceid), for a single [Pricing Unit](#pricingunit) of the associated SKU of a charge. This price is denominated in the [Billing Currency](#billingcurrency). The List Unit Price can be used for calculating savings based on various rate optimization activities. 
```

File: specification/metrics/listunitprice.md [hunk 6]
```diff
@@ -14,7 +14,7 @@ List Unit Price
 
 ## Description
 
-Represents the suggested provider published price for a single pricing measurement unit, which incorporates volume/tier-based reductions while excluding any negotiated or commitment-based discounts.
+A suggested provider-published unit price for a single pricing measurement unit (identified by a [Sku Price Id](#skupriceid)), which incorporates free-tier, volume/tier-based, BYOL-based and dynamically priced SKU rates, while excluding any other discounts or the amortization of upfront charges (one-time or recurring).
```
Comment (udam-f2) [inline specification/metrics/listunitprice.md:17]:
```suggestion
A suggested provider-published unit price, identified by the [SKU Price ID](#skupriceid), for a single [Pricing Unit](#pricingunit) of the associated SKU of a charge.
```

File: specification/metrics/listunitprice.md (hunk 5 above)
Comment (flanakin) [inline specification/metrics/listunitprice.md:3]:
```suggestion
The List Unit Price represents a suggested provider-published unit price for a single pricing measurement unit (identified by a [Sku Price Id](#skupriceid)), which incorporates free-tier, volume/tier-based, BYOL-based and dynamically-priced SKU rates, while excluding any other discounts or the amortization of upfront charges (one-time or recurring). This price is denominated in the [Billing Currency](#billingcurrency). The List Unit Price can be used for calculating savings based on rate optimization activities. In this context, activities such as increased consumption leading to lower volume/tier-based rates, the use of pre-owned software licenses (BYOL - Bring Your Own License), leveraging interruptible resources and/or services, or optimizing usage to take advantage of dynamic pricing models are not considered rate optimization activities.
```

Comment (flanakin) [reply specification/metrics/listunitprice.md:3]: I would rather not make a statement about what is or isn't rate optimization.  Would this convey the same point without taking a stance on what rate optimization is?

```
List Unit Price cannot be used to calculate savings based on volume/tier-based pricing, the use of pre-owned software licenses (BYOL - Bring Your Own License), leveraging interruptible resources and/or services, or optimizing usage to take advantage of dynamic pricing models.
```

Comment (flanakin) [reply specification/metrics/listunitprice.md:3]: ...or what Udam suggested 🙂

Comment (flanakin) [reply specification/metrics/listunitprice.md:3]: Ignore this if you go with Udam's suggestion. Just wanted to catch the missing hyphen. He removed that part.

File: specification/metrics/listunitprice.md [hunk 7]
```diff
@@ -1,8 +1,8 @@
...
 
-The ListUnitPrice column MUST be present in the billing data. This column MUST be a numeric value of type Decimal within the range of non-negative decimal values. It MUST NOT contain null values in cases where the [ChargeType](#chargetype) is 'usage' or 'purchase'. When ListUnitPrice is not null, multiplying ListUnitPrice by [QuantityInPricingUnit](#quantityinpricingunit) MUST equal [ListCost](#listcost).
+The ListUnitPrice column MUST be present in the billing data. This column MUST be a numeric value of type Decimal within the range of non-negative decimal values. It MUST NOT contain null if SkuPriceId is not null and it MUST be null if SkuPriceId is null. When ListUnitPrice is not null, multiplying ListUnitPrice by [QuantityInPricingUnit](#quantityinpricingunit) MUST equal [ListCost](#listcost).
```
Comment (flanakin) [inline specification/metrics/listunitprice.md:5]: Aligning with others and adding the billing currency requirement. (If it's not here, then it's not a requirement 😉)
```suggestion
The ListUnitPrice column MUST be present in the billing data. This column MUST be a numeric value of type Decimal within the range of non-negative decimal values and specified in the BillingCurrency. ListUnitPrice MUST NOT be null or empty if SkuPriceId is not null and MUST be null if SkuPriceId is null. When ListUnitPrice is not null, multiplying ListUnitPrice by [QuantityInPricingUnit](#quantityinpricingunit) MUST equal [ListCost](#listcost).
```

File: specification/metrics/listunitprice.md (hunk 6 above)
Comment (flanakin) [reply specification/metrics/listunitprice.md:17]: I would also remove "A" in the beginning. It's not needed.

File: supporting_content/metrics/listunitprice.md [hunk 8]
```diff
@@ -6,72 +6,71 @@ Current column mappings found in available data sets:
...
 
 ## Discussion / Scratch space
 
-### Current data sources
+### Prices and Currencies
 
-- While Microsoft and AWS offer candidate columns for mapping to ListUnitPrice within billing data, relying on pricing data for all providers is advisable due to mentioned discrepancies and volume/tier-based pricing concerns. Filtering pricing data based on resolved billing data to identify that specific list unit price is possible but often presents a challenge, particularly with volume/tier-based prices.
-- MSFT lists the Retail Price as well as the Unit Price.
+* We deliberated on whether to specify both BillingCurrency and PricingCurrency and decided to enforce a single currency, specifically BillingCurrency. This approach provides consistency and simplifies invoice reconciliation. Since some providers don't include List Unit Prices in BillingCurrency in their public price sheets, the inclusion of a CurrencyExchangeRate dimension in the billing data becomes imperative (name subject to change). This is necessary to ensure we can accurately compare and match ListUnitPrices provided in billing data with those published in public price sheets.
```
Comment (flanakin) [inline supporting_content/metrics/listunitprice.md:50]: This kind of hides the major point here... FOCUS data will NOT be consistent with pricing APIs or invoices, which show prices in the pricing currency.

File: specification/metrics/listunitprice.md [hunk 9]
```diff
@@ -1,8 +1,8 @@
...
+The List Unit Price represents a suggested provider-published unit price for a single pricing measurement unit (identified by a [Sku Price Id](#skupriceid)), which incorporates free-tier, volume/tier-based, BYOL-based and dynamically priced SKU rates, while excluding any other discounts or the amortization of upfront charges (one-time or recurring). This price is denominated in the [Billing Currency](#billingcurrency). The List Unit Price can be used for calculating savings based on rate optimization activities. In this context, activities such as increased consumption leading to lower volume/tier-based rates, the use of pre-owned software licenses (BYOL - Bring Your Own License), leveraging interruptible resources and/or services, or optimizing usage to take advantage of dynamic pricing models are not considered rate optimization activities.
```
Comment (flanakin) [inline specification/metrics/listunitprice.md:4]: We need an explicit statement about the pricing currency issue in all price columns. This is too big of an issue to brush under the rug.
```suggestion

While FOCUS includes all prices in the BillingCurrency, providers may use a separate currency for pricing and invoicing. This means prices in FOCUS may not be the same as what is in the provider's cost and usage data, price list, or invoice. Practitioners need to account for the provider's exchange rate when comparing prices in FOCUS to other sources.

```

**Review by flanakin — CHANGES_REQUESTED** [review]

File: specification/metrics/listunitprice.md (hunk 5 above)
Comment (marc-perreaut) [inline specification/metrics/listunitprice.md:3]: >   a suggested provider-published unit price for a single pricing measurement unit (identified by a [Sku Price Id](#skupriceid))

I am not clear why the single pricing measurement unit is identified by the SKU Price ID. My understanding is that a SKU may have SKU Prices,. which are all identified with different SKU Price IDs:
- List Price (what we are discussing here) = public on-demand price (no commitment)
- Negotiated Price = negotiated on-demand price (no commitment)
- Effective Price  = Negotiated price with commitment-based discount

So I would propose:

>   a suggested provider-published unit price for *the* single pricing measurement unit *that is billed in the charge*

Comment (marc-perreaut) [reply specification/metrics/listunitprice.md:3]: Or maybe the "(identified by a [Sku Price Id])" means there must be a SKU Price ID for the list price?
If yes, it may be confusing, the SKU Price ID of the list price may be different from the SKU Price ID of the record.

To make it clearer, maybe:

> If list price applies from the charge, then the SKU Price ID of the charge must identify a provider-published unit price

File: specification/metrics/listunitprice.md (hunk 9 above)
Comment (flanakin) [reply specification/metrics/listunitprice.md:4]: Will cover this in a separate PR

**Review by flanakin — APPROVED** [review]

File: specification/metrics/listunitprice.md (hunk 5 above)
Comment (ijurica) [reply specification/metrics/listunitprice.md]: to be noted in the Supporting content

File: specification/metrics/listunitprice.md [hunk 10]
```diff
@@ -1,8 +1,8 @@
...
 
-The ListUnitPrice column MUST be present in the billing data. This column MUST be a numeric value of type Decimal within the range of non-negative decimal values. It MUST NOT contain null values in cases where the [ChargeType](#chargetype) is 'usage' or 'purchase'. When ListUnitPrice is not null, multiplying ListUnitPrice by [QuantityInPricingUnit](#quantityinpricingunit) MUST equal [ListCost](#listcost).
+The ListUnitPrice column MUST be present in the billing data. This column MUST be a numeric value of type Decimal within the range of non-negative decimal values and specified in the BillingCurrency. ListUnitPrice MUST NOT be null or empty if SkuPriceId is not null and MUST be null if SkuPriceId is null. When ListUnitPrice is not null, multiplying ListUnitPrice by [QuantityInPricingUnit](#quantityinpricingunit) MUST equal [ListCost](#listcost).
```
Comment (ijurica) [inline specification/metrics/listunitprice.md:5]:
```suggestion
The ListUnitPrice column MUST be present in the billing data. This column MUST be a numeric value of type Decimal within the range of non-negative decimal values and denominated in the BillingCurrency. ListUnitPrice MUST NOT be null or empty if SkuPriceId is not null and MUST be null if SkuPriceId is null. When ListUnitPrice is not null, multiplying ListUnitPrice by [QuantityInPricingUnit](#quantityinpricingunit) MUST equal [ListCost](#listcost).
```

File: specification/metrics/listunitprice.md [hunk 11]
```diff
@@ -1,8 +1,8 @@
...
 
-The ListUnitPrice column MUST be present in the billing data. This column MUST be a numeric value of type Decimal within the range of non-negative decimal values. It MUST NOT contain null values in cases where the [ChargeType](#chargetype) is 'usage' or 'purchase'. When ListUnitPrice is not null, multiplying ListUnitPrice by [QuantityInPricingUnit](#quantityinpricingunit) MUST equal [ListCost](#listcost).
+The ListUnitPrice column MUST be present in the billing data. This column MUST be a numeric value of type Decimal within the range of non-negative decimal values and denominated in the BillingCurrency. ListUnitPrice MUST NOT be null or empty if SkuPriceId is not null and MUST be null if SkuPriceId is null. When ListUnitPrice is not null, multiplying ListUnitPrice by [QuantityInPricingUnit](#quantityinpricingunit) MUST equal [ListCost](#listcost).
```
Comment (ijurica) [inline specification/metrics/listunitprice.md:5]: Question: Do we need to mention the Billing Currency in normative paragraphs of cost columns as well?

File: specification/metrics/listunitprice.md (hunk 5 above)
Comment (ijurica) [reply specification/metrics/listunitprice.md]: One SKU ID can have multiple SKU Price IDs, multiple ListUnitPrices in case of volume/tiered-based rating. 
E.g.:

- tier 1: 0 - 1000 requests; 1.00 $
- tier 2: 1001 - 10000 requests; 0.90$
- etc.

Comment (ijurica) [reply specification/metrics/listunitprice.md]: Going with Udam's suggestion for the definition.
Adding a note in the supporting content about free-tier usage, volume/tier-based pricing, BYOL-based pricing, dynamically-priced SKU rates, etc.

Comment (ijurica) [reply specification/metrics/listunitprice.md]: noted in the supporting content.

**Review by cnharris10 — APPROVED** [review]

File: supporting_content/appendix/quantity_driven_pricing_strategies.md [hunk 12]
```diff
@@ -60,7 +60,8 @@ For better comprehension, please refer to the sample price-tiers configuration a
 | GCP       | Flat-rate based pricing                                                            | Not available                        | *TODO: Add sample data* |
```
Comment (salpaysenturus) [inline supporting_content/appendix/quantity_driven_pricing_strategies.md:60]: I propose that this PR not get merged until either 1) sample data is added to the `BilledUnitPrice Pattern` column and the TODOs are removed, or 2) the column is struck.  I recognize that that column is not strictly related to this PR -- those TODOs precede the changes -- but as long as the doc is being touched, I feel that that should be addressed.

**Review by salpaysenturus — CHANGES_REQUESTED** [review]

File: supporting_content/appendix/quantity_driven_pricing_strategies.md (hunk 12 above)
Comment (flanakin) [reply supporting_content/appendix/quantity_driven_pricing_strategies.md:60]: Note this one is already committed. That said, seems fine to remove something we haven't done yet 😜 I'll make a separate suggestion to simplify the applying it...

Comment (ijurica) [reply supporting_content/appendix/quantity_driven_pricing_strategies.md]: As mentioned in one of the PricingQuantity discussions, this appendix needs a bit more attention. 
I'm unsure if we'll be able to polish everything as planned. That being said, **removing traces of the scoped-out BilledUnitPrice column is a must.**

Comment (ijurica) [reply supporting_content/appendix/quantity_driven_pricing_strategies.md]: The mentions of BilledUnitPrice have been removed from this appendix via the [ListUnitPrice review PR](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/234), so I'm closing this conversation.

**Comment by ijurica** [comment]
TODO: Adjust the definition taking into account that providers publish not only on-demand prices but also unit prices inclusive of commitment-based discounts

**Comment by ijurica** [comment]
> TODO: Adjust the definition taking into account that providers publish not only on-demand prices but also unit prices inclusive of commitment-based discounts


- The ListUnitPrice - SkuPriceId association has been removed.
- The new version of the definition explicitly addresses the discounts.

**Review by marcin-walkow — APPROVED** [review]

File: supporting_content/metrics/listunitprice.md [hunk 13]
```diff
@@ -6,72 +6,79 @@ Current column mappings found in available data sets:
...
+
 ### Current data sources
 
-- While Microsoft and AWS offer candidate columns for mapping to ListUnitPrice within billing data, relying on pricing data for all providers is advisable due to mentioned discrepancies and volume/tier-based pricing concerns. Filtering pricing data based on resolved billing data to identify that specific list unit price is possible but often presents a challenge, particularly with volume/tier-based prices.
-- MSFT lists the Retail Price as well as the Unit Price.
+* For GCP and OCI, Pricing Data serves as the sole data source for ListUnitPrice, while for AWS, it is the preferred data source due to concerns related to volume/tier-based pricing. To determine the relevant price in Pricing Data for a specific charge record, the billing data must include the UnitPriceId. As previously mentioned, when BillingCurrency and PricingCurrency differ, the exchange rate must also be considered.
```
Comment (marc-perreaut) [inline supporting_content/metrics/listunitprice.md:58]: > the billing data must include the UnitPriceId.

I understood the final name for UnitPriceId is actually SKUPriceId.

Comment (ijurica) [inline supporting_content/metrics/listunitprice.md:58]:
```suggestion
* For GCP and OCI, Pricing Data serves as the sole data source for ListUnitPrice, while for AWS, it is the preferred data source due to concerns related to volume/tier-based pricing. To determine the relevant price in Pricing Data for a specific charge record, the billing data must include the SkuPriceId. As previously mentioned, when BillingCurrency and PricingCurrency differ, the exchange rate must also be considered.
```

Comment (ijurica) [reply supporting_content/metrics/listunitprice.md:58]: Thank you for pointing this out! The note refers to a discussion that took place long before the SkuPriceId column was named and specified. An update is coming soon.

File: specification/metrics/listunitprice.md [hunk 14]
```diff
@@ -1,8 +1,8 @@
 # List Unit Price
 
-The List Unit Price represents the suggested provider published unit price for a single pricing measurement unit, which incorporates volume/tier-based rates while excluding any negotiated or commitment-based discounts. This price is denominated in the [Billing Currency](#billingcurrency). It is used to calculate List Cost and savings based on rate optimization activities.
+The List Unit Price represents a suggested provider-published unit price for a single [Pricing Unit](#pricingunit) of the associated SKU, exclusive of any discounts. This price is denominated in the [Billing Currency](#billingcurrency). The List Unit Price can be used for calculating savings based on various rate optimization activities.
```
Comment (udam-f2) [inline specification/metrics/listunitprice.md:3]:
```suggestion
The List Unit Price represents the suggested provider-published unit price for a single [Pricing Unit](#pricingunit) of the associated SKU, exclusive of any discounts. This price is denominated in the [Billing Currency](#billingcurrency). The List Unit Price can be used for calculating savings based on various rate optimization activities.
```

File: specification/metrics/listunitprice.md [hunk 15]
```diff
@@ -14,7 +14,7 @@ List Unit Price
 
 ## Description
 
-Represents the suggested provider published price for a single pricing measurement unit, which incorporates volume/tier-based reductions while excluding any negotiated or commitment-based discounts.
+A suggested provider-published unit price for a single [Pricing Unit](#pricingunit) of the associated SKU, exclusive of any discounts.
```
Comment (udam-f2) [inline specification/metrics/listunitprice.md:17]:
```suggestion
The suggested provider-published unit price for a single [Pricing Unit](#pricingunit) of the associated SKU, exclusive of any discounts.
```

**Review by udam-f2 — APPROVED** [review]

File: supporting_content/metrics/listunitprice.md (hunk 13 above)
Comment (ijurica) [reply supporting_content/metrics/listunitprice.md]: supporting content updated (a bit more details provided).
resolving the conversation

**Review by udam-f2 — APPROVED** [review]
LGTM

File: supporting_content/appendix/quantity_driven_pricing_strategies.md (hunk 12 above)
Comment (udam-f2) [reply supporting_content/appendix/quantity_driven_pricing_strategies.md]: Resolving. This is supporting content that we're not planning to bring into the spec till post v1.0.

File: supporting_content/appendix/quantity_driven_pricing_strategies.md [hunk 16]
```diff
@@ -45,26 +45,27 @@ For better comprehension, please refer to the sample price-tiers configuration a
...
-| OCI       | Flat-rate based pricing                                                            | Not available                                                         | _TODO: Add sample data_ |
-| OCI       | Quantity-dependent pricing                                                         | Not available                                                         | _TODO: Add sample data_ |
+| Provider  | Scenario                                                                           | ListUnitPrice Pattern                                                 |
+| :-------- |:---------------------------------------------------------------------------------- | :-------------------------------------------------------------------- |
+| AWS       | Flat-rate based pricing<br>SKU: E9YHNFENF4XQBZR6                                   | pricing/publicOnDemandRate: 0.000005                                  |
+| AWS       | Quantity-dependent pricing<br> ??? _TODO: look for a higher tier sample_           | pricing/publicOnDemandRate: ??? _TODO: look for a higher tier sample_ |
```
Comment (salpaysenturus) [inline supporting_content/appendix/quantity_driven_pricing_strategies.md:59]: Would be fantastic to go to market in 1.0 without this "TODO" here.

File: supporting_content/appendix/quantity_driven_pricing_strategies.md [hunk 17]
```diff
@@ -45,26 +45,27 @@ For better comprehension, please refer to the sample price-tiers configuration a
 
 #### Sample Billing data
 
-| Scenario | QuantityInPricingUnit | PricingUnit | ListUnitPrice | NegotiatedUnitPrice | BilledUnitPrice | PricingCurrency | BillingCurrency | ListCost | BilledCost | EffectiveCost |
-| :------- | --------------------: | :---------- | ------------: | ------------------: | --------------: | :-------------- | :-------------- | -------: | ---------: | ------------: |
-| S-1      |                    10 | 1GB         |             1 |                0.90 |            0.90 | USD             | USD             |       10 |          9 |             9 |
-| S-1      |                     2 | 1GB         |          0.50 |                0.45 |            0.45 | USD             | USD             |        1 |        0.9 |           0.9 |
-| S-2      |                    12 | 1GB         |          0.50 |                0.45 |            0.45 | USD             | USD             |        6 |        5.4 |           5.4 |
+| Scenario | PricingQuantity | PricingUnit | ListUnitPrice | NegotiatedUnitPrice (1.0+) | PricingCurrency (1.0+) | BillingCurrency | ListCost | BilledCost | EffectiveCost |
```
Comment (salpaysenturus) [inline supporting_content/appendix/quantity_driven_pricing_strategies.md:48]: As a FOCUS contributor, I know that `(1.0+)` means "these columns, although important to this example, aren't making the FOCUS 1.0 spec" -- but as a general audience member, these parentheticals without any further context are confusing to me.  Some verbiage above the table, even a single sentence, that explains these would be helpful.

File: supporting_content/metrics/listunitprice.md [hunk 18]
```diff
@@ -6,72 +6,85 @@ Current column mappings found in available data sets:
...
+| AWS       | CUR;<br>AWS Price List API                         | the closest match - pricing/publicOnDemandRate: The public On-Demand Instance rate in this billing period for the specific row of usage. If you have SKUs with multiple On-Demand public rates, the equivalent rate for the highest tier is displayed. For example, services offering free-tiers or tiered pricing.<br>_Note: Since currently only pricing/publicOnDemandRate definition includes note regarding highest tier, I'm not sure if volume-based pricing applies to all unit prices (rates) in CUR files. If it's not consistent, we might have to rely on the AWS Price List API._ |
```
Comment (salpaysenturus) [inline supporting_content/metrics/listunitprice.md:9]: I'm not sure how formal/informal we can be in these supporting content docs -- but the use of first and second person here (i.e. "I" and "you") feels off to me.  I would suggest using whatever convention we have for this -- and, if we lack any such convention, sidestepping any use of first and second person with some combination of nouns and passive voice would be preferable to me.

```suggestion
| AWS       | CUR;<br>AWS Price List API                         | the closest match - pricing/publicOnDemandRate: The public On-Demand Instance rate in this billing period for the specific row of usage. If there are SKUs with multiple On-Demand public rates, the equivalent rate for the highest tier is displayed. For example, services offering free-tiers or tiered pricing.<br>_Note: Since currently only pricing/publicOnDemandRate definition includes note regarding highest tier, it is unclear whether volume-based pricing applies to all unit prices (rates) in CUR files. If it's not consistent, the practitioner might have to rely on the AWS Price List API._ |
```

**Review by salpaysenturus — APPROVED** [review]
I provided a few more comments that I would prefer to see resolved, but I'm comfortable with going to market with the current draft if those changes do not occur.  Nice work @ijurica!

File: supporting_content/appendix/quantity_driven_pricing_strategies.md (hunk 16 above)
Comment (ijurica) [reply supporting_content/appendix/quantity_driven_pricing_strategies.md:61]: As this pending TODO is part of supporting content, we are proceeding with the merge. However, it is acknowledged that there is a requirement to acquire and provide a higher-tier sample before advancing to market with the specification.
